internal/nlutil/nlutil_linux.go

```diff
@@ -0,0 +1,141 @@
+// Package nlutil wraps vishvandanda/netlink functions that may return EINTR.
+//
+// The wrapped functions currently return EINTR when NLM_F_DUMP_INTR is, meaning
+// something changed during the dump so results may be incomplete or inconsistent.
+//
+// At present, the possibly incomplete/inconsistent results are not returned
+// by netlink functions along with the EINTR. So, it's not possible to do
+// anything but retry.
+//
+// These wrappers will make maxAttempts before returning the EINTR.
+package nlutil
+
+import (
+	"context"
+	"errors"
+
+	"github.com/containerd/log"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+)
+
+// Arbitrary limit on max attempts at netlink calls if they are repeatedly interrupted.
+const maxAttempts = 5
+
+func retryOnIntr(f func() error) {
+	for attempt := 0; attempt < maxAttempts; attempt += 1 {
+		if err := f(); !errors.Is(err, unix.EINTR) {
+			return
+		}
```

Comment:
Silly question; any other error is intentionally swallowed, correct?

Reply:
Swallowed here in the retry loop, yes ... but also assigned in the caller's closure, and returned from this package from there.

Reply:
doh! I see it now 🙈